### PR TITLE
Adding Organism Image Field

### PR DESCRIPTION
**Pull request by spficklin**
# Tripal 4 Core Dev Task

## Description

Adds an orgnanism image field to the organism content type.  This PR adds the field which uses the Drupal image field and also adds it to the default layout of the organism.  

## Testing?

- On a new tripal site:
   1.  Create an oraganism and add as many images as you like.  Upon saving, you should see  the images on the page.  
   2.  Go to `Tripal` > `Page Structure` and click on "manage display" under the "organism' content type.  Click the 'Apply Tripal Layout' button.   Now, return to the organism page and the images should be in a new details box named "Images"
- On an existing tripal site
   1.  Pull the code from this branch
   2. Go to `Tripal` > `Configuration` and click on 'Tripal Field Collection'. 
   3. Delete the collection named "Default Chado Fields"
   4. Clear the cache
   5. Re-import the field collection by going to `Tripal` > `Page Structure`  and click "Import Type Collection".  Check the box labeled "General Content Types (Chado)".  Run the job. You should now have an organism image field.  
   6. Create or edit an oraganism and add as many images as you like.  Upon saving, you should see  the images on the page.  
   2.  Go to `Tripal` > `Page Structure` and click on "manage display" under the "organism' content type.  Click the 'Apply Tripal

### CODE REVIEW

**Comment by spficklin** [comment]
Well dang. It's not passing tests.  Fixing now...

File: tripal_chado/config/install/tripal.tripalfield_collection.general_chado.yml [hunk 1]
```diff
@@ -235,6 +234,58 @@ fields:
...
+        description: One or more images for the organism.
+        cardinality: -1
+        required: false
+        storage_settings:
+            storage_plugin_id: drupal_sql_storage
+            storage_plugin_settings: 
+                base_table: organism
+        settings:
+            termIdSpace: IAO
+            termAccession: "0000101"
```
Comment (dsenalik) [inline tripal_chado/config/install/tripal.tripalfield_collection.general_chado.yml:250]: Just a comment, I have been thinking for a while about creating a field for the `chado.eimage` table. On CarrotOmics I have a Tripal 3 field to do this and use this very term for the field. Would it be a conflict to use the same term as this field uses? (I now see there is much discussion about this in issue #1637)

**Review by dsenalik — CHANGES_REQUESTED** [review]
I tested as described. I could add and remove images as expected.
A note, by accident I added the same file twice and this was allowed. I think this is fine though, and it's probably a Drupal thing anyway.

I did note something else. On a new site after applying layout, the organism appears as
2024-06-23_newsite

When I did the existing site test, I also tried applying the layout, and in this case the organism field is not inside a group.
2024-06-23_oldsite
I suspect the tripal layout YAML needs to be reloaded, so I wonder if this PR needs an update hook?

And tests are not passing 😆

**Comment by spficklin** [comment]
Oh, good point about needing an update hook.   In the testing instructions I gave I walk you through reloading the YAML but we can't expect a user to do that on an existing site. I'll add an update hook.

Regarding the term used for this field.  I think we need to talk more on it.  We will probably want to use this term for any image for any content type.  I'm not sure if that's a problem or not. It is an image.  Does the term need to be more specific (e.g. organism image).  I think we need some conversation about that.

**Comment by laceysanderson** [comment]
This PR is causing schema issues and brought to light the issue described in #1999. Since this PR (1) has very little code, (2) has been open for 1/2 a year and (3) is now dependant on the linked issue which is not easy to fix, I think it should just be closed at this point. We can revisit adding an organism field to these collections at a later point if there is enough demand.

**Pull request closed without merging**